test/dlc_tests/sanity/test_pre_release.py

```diff
@@ -0,0 +1,204 @@
+    tilda = _run_cmd_on_container(container_name, ctx, "ls -A ~")
+    _assert_artifact_free(tilda, stray_artifacts)
+    root = _run_cmd_on_container(container_name, ctx, "ls -A /")
+    _assert_artifact_free(root, stray_artifacts)
+
+
+def test_python_version(image):
+    """
+    Check that the python version in the image tag is the same as the one on a running container.
+
+    :param image: ECR image URI
+    """
+    ctx = Context()
+    container_name = f"py-version-{image.split('/')[-1].replace('.', '-').replace(':', '-')}"
+
+    py_version = ""
+    for tag_split in image.split('-'):
+        if tag_split.startswith('py'):
+            py_version = f"Python {tag_split[2]}.{tag_split[3]}"
+
+    _start_container(container_name, image, ctx)
+    output = _run_cmd_on_container(container_name, ctx, "python --version")
+
+    container_py_version = output.stdout
+    # Due to py2 deprecation, Python2 version gets streamed to stderr. Python installed via Conda also appears to
+    # stream to stderr, hence the pytorch condition.
+    if "Python 2" in py_version or "pytorch" in image:
+        container_py_version = output.stderr
+
+    assert py_version in container_py_version, f"Cannot find {py_version} in {container_py_version}"
+
+
+def test_ubuntu_version(image):
+    """
+    Check that the ubuntu version in the image tag is the same as the one on a running container.
+
+    :param image: ECR image URI
+    """
+    ctx = Context()
+    container_name = f"ubuntu-version-{image.split('/')[-1].replace('.', '-').replace(':', '-')}"
+
+    ubuntu_version = ""
+    for tag_split in image.split('-'):
+        if tag_split.startswith('ubuntu'):
+            ubuntu_version = tag_split.split("ubuntu")[-1]
+
+    _start_container(container_name, image, ctx)
+    output = _run_cmd_on_container(container_name, ctx, "cat /etc/os-release")
+    container_ubuntu_version = output.stdout
+
+    assert "Ubuntu" in container_ubuntu_version
+    assert ubuntu_version in container_ubuntu_version
+
+
+def test_framework_version_cpu(cpu):
+    """
+    Check that the framework version in the image tag is the same as the one on a running container.
+
+    :param cpu: ECR image URI with "cpu" in the name
+    """
+    image = cpu
+    if "tensorflow-inference" in image:
+        pytest.skip(msg="TF inference does not have core tensorflow installed")
+
+    tested_framework, tag_framework_version = _get_framework_and_version_from_tag(image)
+    ctx = Context()
+    container_name = f"framework-version-{image.split('/')[-1].replace('.', '-').replace(':', '-')}"
+    _start_container(container_name, image, ctx)
+    output = _run_cmd_on_container(
+        container_name, ctx, f"import {tested_framework}; print({tested_framework}.__version__)", executable="python"
+    )
+
+    assert tag_framework_version == output.stdout.strip()
+
+
+@pytest.mark.parametrize("ec2_instance_type", ['p2.xlarge'], indirect=True)
+def test_framework_version_gpu(gpu, ec2_connection):
```

Comment:
I don't think we need to run this test on EC2. Why are separate CPU and GPU tests required to test framework-version?

Reply:
The test errors out when run on the codebuild instance, as the instance does not have nvidia-docker support. Import will fail for gpu optimized framework binaries